L1Trigger/VertexFinder/plugins/VertexAnalyzer.cc

```diff
@@ -0,0 +1,820 @@
+        100,
+        0,
+        100);
+
+    hisNumVxIterations_ = inputDir.make<TH1F>(
+        "hisNumVxIterations_", "Number of Iterations (Vertex Reconstruction); No. Iterations; Entries", 100, 0, 500);
+    hisNumVxIterationsPerTrack_ =
+        inputDir.make<TH1F>("hisNumVxIterationsPerTrack_",
+                            "Number of Iterations per Track(Vertex Reconstruction); No. Iterations; Entries",
+                            100,
+                            0,
+                            200);
+
+    hisCorrelatorInputTracks_ = inputDir.make<TH1F>(
+        "hisCorrelatorInputTracks_", "Number of Input Tracks at L1 correlator; No. L1 Tracks; Entries", 30, 0, 200);
+    hisCorrelatorTPInputTracks_ = inputDir.make<TH1F>(
+        "hisCorrelatorTPInputTracks_", "Number of Input Tracks at L1 correlator; No. L1 Tracks; Entries", 30, 0, 200);
+
+    hisCorrelatorInputVertices_ =
+        inputDir.make<TH1F>("hisCorrelatorInputVertices_",
+                            "Number of Input Vertices at L1 correlator; No. L1TkVertices; Entries",
+                            25,
+                            0,
+                            100);
+    hisCorrelatorTPInputVertices_ =
+        inputDir.make<TH1F>("hisCorrelatorTPInputVertices_",
+                            "Number of Input Vertices at L1 correlator; No. L1TKVertices; Entries",
+                            25,
+                            0,
+                            100);
+
+    std::vector<unsigned int> emptyVector;
+    emptyVector.assign(10, 0);
+  }
+
+  template <typename T>
+  int VertexAnalyzer::getIndex(std::vector<T> collection, const T reference) {
+    int index = -1;
+    auto itr = std::find(collection.begin(), collection.end(), reference);
+    if (itr != collection.end()) {
+      index = std::distance(collection.begin(), itr);
+    }
+    return index;
+  }
+
+  void VertexAnalyzer::beginJob(){};
+
+  void VertexAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
+    // Note useful info about MC truth particles and about reconstructed stubs .
+    InputData inputData(iEvent,
+                        iSetup,
+                        settings_,
+                        hepMCInputTag,
+                        genParticleInputTag,
+                        tpInputTag,
+                        stubInputTag,
+                        stubTruthInputTag,
+                        clusterTruthInputTag);
+
+    edm::Handle<TTTrackCollectionView> l1TracksHandle;
+    iEvent.getByToken(l1TracksToken_, l1TracksHandle);
+
+    std::vector<L1TrackTruthMatched> l1Tracks;
+    l1Tracks.reserve(l1TracksHandle->size());
+    {
+      // Get the tracker geometry info needed to unpack the stub info.
+      edm::ESHandle<TrackerGeometry> trackerGeometryHandle;
+      iSetup.get<TrackerDigiGeometryRecord>().get(trackerGeometryHandle);
+      edm::ESHandle<TrackerTopology> trackerTopologyHandle;
+      iSetup.get<TrackerTopologyRcd>().get(trackerTopologyHandle);
```

Comment:
Please migrate to EventSetup consumes (as pointed to in the clang-tidy warnings https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-b282c5/14852/llvm-analysis/cmsclangtidy.txt)